From b4adbd496e0feda336b5315a5dfca427192a0f79 Mon Sep 17 00:00:00 2001
From: Rajan Vaja <rajan.vaja@xilinx.com>
Date: Mon, 13 Jan 2020 00:20:48 -0800
Subject: [PATCH 0810/1852] soc: xilinx: vcu: Move structure definition to
 header file

commit 63cb1ddc163c9d9325f61e840fd37652a8e2c83b from
https://github.com/Xilinx/linux-xlnx.git

Move xvcu_device structure definition to header file
as it will be used by other vcu files.

Signed-off-by: Rajan Vaja <rajan.vaja@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/soc/xilinx/xlnx_vcu.c | 18 ------------------
 include/soc/xilinx/xlnx_vcu.h | 18 +++++++++++++++++-
 2 files changed, 17 insertions(+), 19 deletions(-)

diff --git a/drivers/soc/xilinx/xlnx_vcu.c b/drivers/soc/xilinx/xlnx_vcu.c
index 0451d9fc2d5b..0ed301d2dfd6 100644
--- a/drivers/soc/xilinx/xlnx_vcu.c
+++ b/drivers/soc/xilinx/xlnx_vcu.c
@@ -96,24 +96,6 @@
 #define FRAC				100
 #define LIMIT				(10 * MHZ)
 
-/**
- * struct xvcu_device - Xilinx VCU init device structure
- * @dev: Platform device
- * @pll_ref: pll ref clock source
- * @aclk: axi clock source
- * @logicore_reg_ba: logicore reg base address
- * @vcu_slcr_ba: vcu_slcr Register base address
- * @coreclk: core clock frequency
- */
-struct xvcu_device {
-	struct device *dev;
-	struct clk *pll_ref;
-	struct clk *aclk;
-	void __iomem *logicore_reg_ba;
-	void __iomem *vcu_slcr_ba;
-	u32 coreclk;
-};
-
 /**
  * struct xvcu_pll_cfg - Helper data
  * @fbdiv: The integer portion of the feedback divider to the PLL
diff --git a/include/soc/xilinx/xlnx_vcu.h b/include/soc/xilinx/xlnx_vcu.h
index 9a226c634569..3e8b77a8817a 100644
--- a/include/soc/xilinx/xlnx_vcu.h
+++ b/include/soc/xilinx/xlnx_vcu.h
@@ -10,7 +10,23 @@
 #ifndef _XILINX_VCU_H_
 #define _XILINX_VCU_H_
 
-struct xvcu_device;
+/**
+ * struct xvcu_device - Xilinx VCU init device structure
+ * @dev: Platform device
+ * @pll_ref: pll ref clock source
+ * @aclk: axi clock source
+ * @logicore_reg_ba: logicore reg base address
+ * @vcu_slcr_ba: vcu_slcr Register base address
+ * @coreclk: core clock frequency
+ */
+struct xvcu_device {
+	struct device *dev;
+	struct clk *pll_ref;
+	struct clk *aclk;
+	void __iomem *logicore_reg_ba;
+	void __iomem *vcu_slcr_ba;
+	u32 coreclk;
+};
 
 u32 xvcu_get_color_depth(struct xvcu_device *xvcu);
 u32 xvcu_get_memory_depth(struct xvcu_device *xvcu);
-- 
2.31.1

